lib/ansible/modules/notification/teams.py

```diff
@@ -0,0 +1,151 @@
+        To add links to text, developers **must** include Markdown syntax for
+        hyperlinks.
+    required: true
+  color:
+    description:
+      - You can also control a color bar by setting color to a color hex code.
+        This bar appears to the left of the cards content, and is primarily
+        used to indicate status to the user.
+    required: false
+    default: None
+  url:
+    description:
+      - The webhook URL is given to you when you create a new Connector in
+        Microsoft Teams. This URL is used to communicate with the target
+        Office 365 Group.
+        All the GUIDs within the URL define the reference to the context of the
+        current webhook.
+    required: true
+"""
+
+EXAMPLES = """
+- name: Send notification message via Microsoft Teams
+  teams:
+    url: https://outlook.office.com/webhook/GUID/IncomingWebhook/GUID/GUID
+    text: 'Deployment completed on {{ inventory_hostname }}'
+  delegate_to: localhost
+
+- name: Send notification message via Microsoft Teams
+  teams:
+    url: https://outlook.office.com/webhook/GUID/IncomingWebhook/GUID/GUID
+    title: 'Ansible status'
+    text: '{{ inventory_hostname }} is up and running'
+    color: '00CC00'
+  delegate_to: localhost
+"""
+
+RETURN = """
+msg:
+  description: OK or error message with payload
+  returned: success
+  type: string
+  sample: "OK"
+"""
+
+# import module snippets
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.urls import fetch_url
+
+
+def build_payload_for_teams(module, text, title, color):
+    payload = dict(text=text)
+
+    if title is not None:
+        payload['title'] = title
+
+    if color is not None:
+        payload['themeColor'] = color
+
+    payload=module.jsonify(payload)
+    return payload
+
+
+def do_notify_teams(module, teams_incoming_webhook, payload):
+    headers = {
+        'Content-Type': 'application/json',
+    }
+    response, info = fetch_url(
+        module=module,
+        url=teams_incoming_webhook,
+        headers=headers,
+        method='POST',
+        data=payload
+    )
+
+    if info['status'] != 200:
+        module.fail_json(
+            msg=" failed to send %s to Teams incoming webhook: %s" % (
+                payload,
+                info['msg']
+            )
+        )
+
+def main():
+
+    module = AnsibleModule(
+        argument_spec = dict(
+            url = dict(type='str', required=True, no_log=True),
+            text = dict(type='str', required=True),
+            title = dict(type='str', required=False, default=None),
+            color = dict(type='str', required=False, default=None)
+        )
+    )
+
+    url = module.params['url']
+    text = module.params['text']
+    title = module.params['title']
+    color = module.params['color']
+
+    payload = build_payload_for_teams(module, text, title, color)
+    do_notify_teams(module, url, payload)
+
+    module.exit_json(msg="OK")
```

Comment:
What's the purpose of sending `msg="OK"` if the success of the task indicates the same thing ?